bson/src/main/org/bson/BasicBSONEncoder.java

```diff
@@ -152,10 +152,10 @@ protected void _putObjectField(final String name, final Object initialValue) {
             putNull(name);
         } else if (value instanceof Date) {
             putDate(name, (Date) value);
-        } else if (value instanceof Number) {
-            putNumber(name, (Number) value);
         } else if (value instanceof Decimal128) {
             putDecimal128(name, (Decimal128) value);
+        } else if (value instanceof Number) {
```

Comment:
Since Decimal128 is-a Number now, have to move its handling above the generic Number handling

Reply:
Ack

Reply:
Yeah, otherwise I wouldn't have seen it.  After I fixed this one I searched in IntelliJ for all usage of the Number interface just to make sure nothing else was there.  That's how I found the NumberCodecHelper.

Reply:
Ack